web-components/src/components/table/ActionsTable.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Table,
+  TableBody,
+  TableHead,
+  TableRow,
+  Box,
+  TableFooter,
+} from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import {
+  StyledTableCell,
+  StyledTableContainer,
+  StyledTableRow,
+  getTablePaginationPadding,
+  // StyledTableSortLabel,
+} from './';
+// import {
+//   getComparator,
+//   Order,
+//   stableSort,
+// } from 'web-components/lib/components/table/sort';
+import { TablePagination } from './TablePagination';
+
+const useStyles = makeStyles(theme => ({
+  borderLeft: {
+    // absolutely position border to get around MUI style quirks
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+    borderLeft: `1px solid ${theme.palette.info.light}`,
+  },
+  greyText: {
+    color: theme.palette.info.main,
+  },
+}));
+
+export type renderActionButtonsFunc = (i: number) => React.ReactElement;
```

Comment:
```suggestion
export type RenderActionButtonsFunc = (i: number) => React.ReactElement;
```
I'm a bit confused about what the index is for - just to pass an index down to be used as `key`?

Reply:
to get the corresponding row and inform the "action modal" that will open up (eg if I want to retire some ecocredits, we should provide the batch denom and max amount that the user can retire to the modal, and we can get this using `row[i]` info)